handlers/public.go

```diff
@@ -0,0 +1,208 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/payments.api.ch.gov.uk/config"
+	"github.com/companieshouse/payments.api.ch.gov.uk/data"
+
+	"github.com/gorilla/pat"
+)
+
+type createPaymentResource struct {
+	RedirectURI string `json:"redirect_uri"`
+	Reference   string `json:"reference"`
+	Resource    string `json:"resource"`
+	State       string `json:"state"`
+}
+
+// PaymentResource is the payment details to be stored in the DB and returned in the response
+type PaymentResource struct {
+	//	ID                      string    `json:"id"                                  bson:"_id"`                                 // TODO implement
+	Amount                  string    `json:"amount"                              bson:"amount"`                              // TODO implement
```

Comment:
There should be two models to represent the data (one matching the spec for the data coming in the request and one matching the data in the database).
```
request -> spec model -> transformer -> DB model -> DB
```

The JSON tags would be in the spec model and the BSON tags would be in the database model.

[Company accounts API](https://github.com/companieshouse/company-accounts.api.ch.gov.uk) contains an example of this.